### DISCUSSION TITLE
Extracting one feature

### DISCUSSION THREAD

**Discussion by atulvermaPRFT** [twilio-professional-services/flex-project-template#134] [Q&A] [answered]
Hi, 
I would like to ask how we can extract one feature from this repo and create a separate plugin for it using twilio flex:plugins:create plugin-sample --install --flexui2  command ?

**Comment by jared-hunter** [comment, marked as answer]
There is no compatibility between this project and the CLI create command and there is not any intention at the moment to create any.

If you wanted to single out a feature you can clone the repository and remove all the other features from the '<plugin>/src/features-library' folder, then remove the references from the '</plugin>/src/flex-hooks' folder and the associated functions from the 'serverless-functions/src/functions/features' folder (recommended and easiest)

Alternatively you can cherry pick any out the feature folder into your own plugin and call on each of the hooks listed under the '<plugin>/src/feature-library/<feature-name>/flex-hooks' and the associated serverless functions.  This may also include the common serverless functions and the common serverless utilities in the plugin '<plugin>/src/utils/serverless/' (this may be a little more time consuming to make sure all dependencies are brought over but would result in only the code you need)
